### DISCUSSION TITLE
Problems with myst via socks5 proxy

### DISCUSSION THREAD

**Discussion by skier789** [engageub/InternetIncome#150] [General]
First of all, thanks to the author for this project

Is there still no way to make myst work through a socks5 proxy?

I've tried everything I can search for and it doesn't work, neither does socks5 support udp

Does anyone have a solution for this?

**Comment by engageub** [comment]
Hello,
Thank you for the feedback. 
Proxy for Mysterium is not supported at the moment due to ongoing issue
You can run Mysterium with proxies in [test](https://github.com/engageub/InternetIncome/tree/test) branch. That was enabled for testing as there is an open issue. it requires Socks5 proxies with UDP support.
It has been disabled in Main branch.
It works with Direct Connection, Multi IP and Multi VPN but not with proxies.

Please see https://github.com/xjasonlyu/tun2socks/issues/262 for more details.


Thank

**Comment by skier789** [comment]
Thanks for the reply.
 test branch is also for testing purposes only and not a solution is it.
That means myst won't run on socks5 proxy at all

![image](https://github.com/user-attachments/assets/d567fd71-264c-4367-bd8d-07d0c9e98689)
The current state of affairs is that you can establish a connection but you can't generate data.

**Reply by engageub** [reply]
Hi,
Test branch was not only meant for testing purposes, but also for advanced users who wish to use Multi IP and Multi VPN.
New Apps get added to test branch first and then to main branch.
From the screenshot, some data is being transferred. 
Please make sure all the services are active or enable each service in different proxy.
This way you will know which service is working based on data transfer. 
You will have to wait upto 24 hours to see the traffic. 
It did not work earlier, but if it does work for you based on different service, do let us know.
Please enable logs in `properties.conf` to check if there are any errors.


Thank you
